src/extension.ts

```diff
@@ -90,32 +88,36 @@ export function activate(context: ExtensionContext) {
 
     const startServer = { command: csBinaryPath, args };
 
-    client = new LanguageClient(
+    let client = new LanguageClient(
       "smithyLsp",
       "Smithy LSP",
       startServer,
       clientOptions
     );
 
+    // Start the client. This will also launch the server.
+    await client.start();
+
     const smithyContentProvider = createSmithyContentProvider(client);
+
+    const registerRestartCommand = vscode.commands.registerCommand(
+      "smithyLsp.restart",
+      () => {
+        client.restart();
+      }
+    );
+
     context.subscriptions.push(
       workspace.registerTextDocumentContentProvider(
         "smithyjar",
         smithyContentProvider
       ),
-      // Start the client. This will also launch the server
-      client.start()
+      client,
```

Comment:
**thought:** so in 8.x.x the client itself is a `Disposable`, and its `dispose` calls `stop`. Subscriptions should deal with the cleanup - this is why I don't think we need to keep the client reference between activate/deactivate.